src/render/scene.cpp

```diff
@@ -61,10 +61,22 @@ MI_VARIANT Scene<Float, Spectrum>::Scene(const Properties &props) {
     for (Sensor *sensor: m_sensors)
         sensor->set_scene(this);
 
-    if constexpr (dr::is_cuda_v<Float>)
-        accel_init_gpu(props);
-    else
-        accel_init_cpu(props);
+    // Decide whether to use acceleration data structures for ray intersections
+    // TODO: do we even want a heuristic? Could lead to surprising changes in performance for the user.
+    bool naive_intersection_desirable = m_shapes.size() <= 5;
```

Comment:
As discussed with @Speierers, we may not want to enable this feature unless the user explicitly asks for it. The downside is that users may not find out it exists, but it's probably better than creating surprising performance jumps due to this heuristic.